src/Npgsql/NpgsqlConnector.cs

```diff
@@ -450,9 +450,74 @@ void WriteStartupMessage(string username)
             if (timezone != null)
                 startupParams["TimeZone"] = timezone;
 
+            var options = Settings.Options?.Length > 0
+                ? Settings.Options
+                : PostgresEnvironment.Options?.Length > 0
+                    ? PostgresEnvironment.Options
+                    : null;
+
+            if (options != null)
+            {
+                var pos = 0;
+                while (pos < options.Length)
+                {
+                    var key = ParseKey(options, ref pos);
+                    var value = ParseValue(options, ref pos);
+                    startupParams[key] = value;
+                }
+            }
+
             WriteStartup(startupParams);
         }
 
+        static string ParseKey(string str, ref int pos)
+        {
+            var start = pos;
+            for (;pos < str.Length; pos++)
+            {
+                if (str[pos] == '=')
+                {
+                    var key = str.Substring(start, pos - start);
+                    pos++;
+                    return key.Length > 0
+                        ? key
+                        : throw new FormatException(
+                            $"Invalid syntax for connection string parameter '{nameof(NpgsqlConnectionStringBuilder.Options)}': Missing key.");
+                }
+            }
+            throw new FormatException($"Invalid syntax for connection string parameter '{nameof(NpgsqlConnectionStringBuilder.Options)}': Missing '='.");
+        }
+
+        static string ParseValue(string str, ref int pos)
+        {
+            var escaped = false;
+            var value = new StringBuilder();
+            for (;pos < str.Length; pos++)
+            {
+                if (escaped)
+                    escaped = false;
+                else if(str[pos] == ' ')
+                {
+                    pos++;
+                    for (; pos < str.Length && str[pos] == ' '; pos++)
+                    {
+                    }
+
+                    break;
+                }
+                else if (str[pos] == '\\')
+                {
+                    escaped = true;
```

Comment:
Do you really need `escaped` here? You could simply  increment `pos` and fall through to appending it below, simplifying the code (or am I missing something?)